lib/sidekiq/antidote/inhibitor.rb

```diff
@@ -50,6 +50,26 @@ def eql?(other)
           && id == other.id && treatment == other.treatment && class_qualifier == other.class_qualifier
       end
       alias == eql?
+
+      def treat(message)
+        case treatment
+        when "postpone"
+          queue = message["queue"] || "default"
+
+          message["enqueued_at"] = Time.now.to_f
+          to_push = Sidekiq.dump_json(message)
+
+          redis("LPUSH", "queue:#{queue}", to_push)
+        when "kill"
+          DeadSet.new.kill(Sidekiq.dump_json(message))
+        end
+      end
+
+      private
+
+      def redis(...)
+        Sidekiq.redis { _1.call(...) }
+      end
```

Comment:
Inhibitor should remain simple data object. We should remove `def lethal?` and in middlewares do something like:

``` ruby
case inhibitor.treatment
when "kill" then ...
...
```

Middlewares have knowledge of which queue it was taken from.

Either way, one of the problems with this treatment - it may cause queue starvation - and useless pop/push loops.